src/Swarm/TUI/View.hs

```diff
@@ -109,18 +111,36 @@ drawUI s
   | otherwise = case s ^. uiState . uiMenu of
     -- We should never reach the NoMenu case if uiPlaying is false; we would have
     -- quit the app instead.  But just in case, we display the main menu anyway.
-    NoMenu -> [drawMainMenuUI (s ^. uiState . appData . at "logo") (mainMenu NewGame)]
-    MainMenu l -> [drawMainMenuUI (s ^. uiState . appData . at "logo") l]
+    NoMenu -> [drawMainMenuUI s (mainMenu NewGame)]
+    MainMenu l -> [drawMainMenuUI s l]
     NewGameMenu stk -> [drawNewGameMenuUI stk]
+    MessagesMenu -> [drawMainMessages s]
     AboutMenu -> [drawAboutMenuUI (s ^. uiState . appData . at "about")]
 
-drawMainMenuUI :: Maybe Text -> BL.List Name MainMenuEntry -> Widget Name
-drawMainMenuUI logo l =
+drawMainMessages :: AppState -> Widget Name
+drawMainMessages s = renderDialog dial . padBottom Max . scrollList . drawLogs $ s ^. runtimeState . eventLog
+ where
+  dial = dialog (Just "Messages") Nothing maxModalWindowWidth
+  scrollList = withVScrollBars OnRight . vBox
+  drawLogs = map (drawLogEntry True) . toList
+
+drawMainMenuUI :: AppState -> BL.List Name MainMenuEntry -> Widget Name
+drawMainMenuUI s l =
   vBox
     [ maybe emptyWidget drawLogo logo
+    , hCenter . padTopBottom 2 $ newVersioWidget version
     , centerLayer . vLimit 5 . hLimit 20 $
-        BL.renderList (const (hCenter . drawMainMenuEntry)) True l
+        BL.renderList (const (hCenter . drawMainMenuEntry s)) True l
     ]
+ where
+  logo = s ^. uiState . appData . at "logo"
+  version = s ^. runtimeState . upstreamRelease
+
+newVersioWidget :: Either NewReleaseFailure String -> Widget n
+newVersioWidget = \case
+  Right ver -> txt $ "New version " <> T.pack ver <> " is available!"
+  Left (OnDevelopmentBranch _b) -> txt "Good luck developing!"
+  Left _ -> txt "The game is up-to-date!"
```

Comment:
There is always the tension between "it will be nice for the user" and "it will be neat for the developer fixing bugs". :laughing:

I will fix that shortly. :wink: